src/neuroglancer/sliceview/png/png_wasm.c

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright 2022 William Silvermsith
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "spng.h"
+#include <stdbool.h>
+
+#define RET(val) spng_ctx_free(ctx); return (val);
+
+int png_decompress(
+	unsigned char* buf, unsigned int num_bytes, 
+    void* out, bool convert_to_grayscale
```

Comment:
Should also pass in the size of the output buffer, even though in this case it will have already been validated by javascript code.